crates/rattler_conda_types/src/channel/mod.rs

```diff
@@ -667,4 +671,41 @@ mod tests {
             "https://prefix.dev/conda-forge"
         );
     }
+
+    #[test]
+    fn compare_with_or_without_backslash() {
+        let channel_config = ChannelConfig {
+            channel_alias: Url::from_str("https://conda.anaconda.org").unwrap(),
+            root_dir: std::env::current_dir().expect("No current dir set"),
+        };
+
+        // Normal channel should have backslash
+        let channel = Channel::from_str("conda-forge", &channel_config).unwrap();
+        let channel_with_backslash =
+            Channel::from_str("https://conda.anaconda.org/conda-forge/", &channel_config).unwrap();
+        assert_eq!(channel.base_url(), channel_with_backslash.base_url());
+
+        // Channel with backslash should have backslash
+        let channel_with_backslash = Channel::from_str("conda-forge/", &channel_config).unwrap();
+        let channel =
+            Channel::from_str("https://conda.anaconda.org/conda-forge", &channel_config).unwrap();
+        assert_eq!(channel.base_url(), channel_with_backslash.base_url());
```

Comment:
Here you are basically checking if for some input when you parse it to a channel it contains a backslash. But there is a lot of duplication. Maybe write a for loop over some input values and in the for loop check if the base_url contains a backslash. You can then also add more input values. Try with a url that already contains a backslash etc.

Reply:
:+1: